File: python/vedant_bhatnagar/DevsnestProblems_dailyUpdated/ValidAnagram.py
def solve(s1, s2):
    # CODE HERE
    dict1 = {}
    dict2 = {}

    for i in s1:
        if i in dict1:
            dict1[i] += 1
        else:
            dict1[i] = 1

    for i in s2:
        if i in dict2:
            dict2[i] += 1
        else:
            dict2[i] = 1

    if dict1 == dict2:
        return 1
    else:
        return 0        

File: java/ankit_kanyal/devnest_problems/phase_1/ValidAnagram.java
// link- https://devsnest.in/algo-challenges/valid-anagram

static int solve(String s1, String s2){
//CODE HERE 
Map<Character,Integer> charCount1=new HashMap<>();
Map<Character,Integer> charCount2=new HashMap<>();
if(s1.length()!=s2.length()) return 0;
for(char c:s1.toCharArray()){
    charCount1.put(c,charCount1.getOrDefault(c,0)+1);
}
for(char c:s2.toCharArray()){
    charCount2.put(c,charCount2.getOrDefault(c,0)+1);
}

for(Map.Entry entry:charCount2.entrySet()){
    Character key=(Character)entry.getKey();
    Integer value=(Integer)entry.getValue();
    // System.out.println("key "+key+" value "+value);
    if(!charCount1.containsKey(key) || charCount1.get(key)<value) return 0;
}
return 1;
}